packages/test-worker/src/test-worker-local.ts

```diff
@@ -0,0 +1,104 @@
+import * as process from 'process';
+import * as path from 'path';
+import { EventEmitter } from 'events';
+import { Sandbox } from '@testring/sandbox';
+import { testAPIController } from '@testring/api';
+import {
+    ITransport,
+    ITestExecutionMessage,
+    ITestExecutionCompleteMessage,
+    TestWorkerAction,
+    TestStatus,
+    TestEvents,
+    ITransportChild,
+    ITransportDirectMessage,
+} from '@testring/types';
+
+export class TestWorkerLocal extends EventEmitter implements ITransportChild {
+
+    constructor(
+        private transportInstance: ITransport,
+    ) {
+        super();
+    }
+
+    public kill() {
+        // @TODO make kill functionality
+    }
+
+    public send(message: ITransportDirectMessage, callback: (error: Error) => void): boolean {
+        const { payload, type } = message;
+
+        if (type === TestWorkerAction.executeTest) {
```

Comment:
early exit?

```
if (type !== TestWorkerAction.executeTest) {
    return true; 
}
```

Reply:
It always return true, cause we are making it ChildProcess like.

> [subprocess.send() ](https://nodejs.org/api/child_process.html#child_process_subprocess_send_message_sendhandle_options_callback) will return false if the channel has closed or when the backlog of unsent messages exceeds a threshold that makes it unwise to send more. Otherwise, the method returns true. The callback function can be used to implement flow control.